adafruit_platformdetect/revcodes.py

```diff
@@ -0,0 +1,296 @@
+        if not self._valid_value(data[index], values):
+            raise ValueError(f"Invalid value {data[index]} for property {name}")
+        if raw:
+            return data[index]
+        return self._format_value(data[index], values)
+
+    @staticmethod
+    def _format_value(value, valid_values):
+        if valid_values is float or valid_values is int:
+            return valid_values(value)
+        return valid_values[value]
+
+    @staticmethod
+    def _valid_value(value, valid_values):
+        if valid_values is float or valid_values is int:
+            return isinstance(value, valid_values)
+        if isinstance(valid_values, (tuple, list)) and 0 <= value < len(valid_values):
+            return True
+        if isinstance(valid_values, dict) and value in valid_values.keys():
+            return True
+        return False
+
+    def _get_property(self, name, raw=False):
+        if name not in NEW_REV_STRUCTURE:
+            raise ValueError(f"Unknown property {name}")
+        if self.is_new_format():
+            return self._get_rev_prop_value(name, raw=raw)
+        if name in OLD_REV_EXTRA_PROPS:
+            return self._get_rev_prop_value(
+                name, structure=OLD_REV_EXTRA_PROPS, raw=raw
+            )
+        return self._get_old_rev_prop_value(name, raw=raw)
+
+    def is_new_format(self):
+        """Check if the code is in the new format"""
+        return self._get_rev_prop_value("rev_style", raw=True) == 1
+
+    @property
+    def overvoltage(self):
+        """Overvoltage allowed/disallowed"""
+        return self._get_property("overvoltage")
+
+    @property
+    def warranty_bit(self):
+        """Warranty bit"""
+        return self._get_property("warranty")
+
+    @property
+    def otp_program(self):
+        """OTP programming allowed/disallowed"""
+        return self._get_property("otp_program")
+
+    @property
+    def otp_read(self):
+        """OTP reading allowed/disallowed"""
+        return self._get_property("otp_read")
+
+    @property
+    def rev_style(self):
+        """Revision Code style"""
+        # Force new style for Rev Style
+        return self._get_rev_prop_value("rev_style")
+
+    @property
+    def memory_size(self):
+        """Memory size"""
+        return self._get_property("memory_size")
+
+    @property
+    def manufacturer(self):
+        """Manufacturer"""
+        return self._get_property("manufacturer")
+
+    @property
+    def processor(self):
+        """Processor"""
+        return self._get_property("processor")
+
+    @property
+    def type(self):
+        """Specific Model"""
+        return self._get_property("type")
+
+    @property
+    def type_raw(self):
+        """Raw Value of Specific Model"""
+        return self._get_property("type", raw=True)
+
+    @property
+    def revision(self):
+        """Revision Number"""
+        return self._get_property("revision")
```

Comment:
You could use a Data Descriptor class to factor these all out. (Works like CP Register does.)

Reply:
I think since this has been well tested as it is and I don't feel confident with descriptors, I'd rather use this as a starting point. That could be a future improvement.